script.js

```diff
@@ -0,0 +1,908 @@
+const runDealCardsEngine = () => {
+  // Change delay in MS depending on whether covercards are present
+  let delayInDrawingCardsAnimation;
+  // Animate out cover-cards first if present
+  if (coverCardShown === true) {
+    coverCardAnimateOut();
+    coverCardShown = false;
+    delayInDrawingCardsAnimation = 800;
+  } else if (coverCardShown === false) {
+    const existHoldStatusDisplay = document.querySelector('.holdStatus');
+    existHoldStatusDisplay.innerText = '';
+    fadeOutCurrHandAnimation();
+
+    setTimeout(() => {
+      cardsContainer.innerText = '';
+      displayCoverCard();
+    }, 1000);
+
+    setTimeout(() => {
+      coverCardAnimateOut();
+    }, 2500);
+
+    delayInDrawingCardsAnimation = 3500;
+  }
+  shuffledDeck = shuffleCards(makeDeck());
+  setTimeout(() => {
+    cardsContainer.innerText = '';
+
+    // clear playerHand first before drawing initial hand
+    playerHand.length = 0;
+
+    // clear cache of previous' hand's win combi
+    nameOfWinCombi = 'No winning hand.\n Please insert credits to continue';
+    drawInitialHand();
+    checkForWinCombi();
+    console.log('creditsLeft', creditsLeft);
+
+    // Calculate the level of payout and allow relevant pointer to scroll
+    calcPayOutLevel();
+
+    // Scroll the top header to display the winning combi
+    topScoreScrollDisplay();
+
+    // reset insert credits and deduct from credits left:
+    creditsLeft -= numCreditsInserted;
+    numCreditsInserted = 0;
+
+    // reset stats display and make a new one;
+    statsDisplay.innerText = '';
+    createGameStatsDisplay();
+  }, delayInDrawingCardsAnimation);
+
+  gameRound += 1;
+};
+
+// Function that creates a button for dealing cards and run corresponding events
+const createDealCardsBtn = () => {
+  const dealBtn = document.createElement('button');
+  dealBtn.classList.add('btn');
+  dealBtn.classList.add('btn-design5');
+  dealBtn.setAttribute('id', 'dealBtn');
+  dealBtn.innerText = 'DEAL';
+  dealBtn.addEventListener('click', () => {
+    console.log(gameRound, 'gameRound');
+    // reset the top scrolling machine displays and recreate it
+
+    if (numCreditsInserted > 0) {
+      if (gameRound > 0) {
+        refreshScrollMachineDisplay();
+      }
+      dealCardsAndAnimateMsg();
+    } else {
+      statsDisplay.innerText = '';
+      createGameStatsDisplay('Please insert \ncredits to deal!');
+      outputMsgDisplay.classList.add('animate__animated');
+      outputMsgDisplay.classList.add('animate__flash');
+    }
+    gameRound += 1;
+  });
+  buttonsContainer.appendChild(dealBtn);
+};
+
+// Function that creates a button that allows user to swap cards
+const createSwapCardsBtn = () => {
+  const swapBtn = document.createElement('button');
+  swapBtn.classList.add('btn');
+  swapBtn.classList.add('btn-design1');
+  swapBtn.setAttribute('id', 'swapBtn');
+  swapBtn.innerText = 'SWAP';
+
+  swapBtn.addEventListener('click', () => {
```

Comment:
Consider decomposing the game/app logic out of the function to create the UI. Separating UI from app logic allows us to test each more independently, improving the maintainability of our applications.

Reply:
E.g. The callback function can be a helper function

Reply:
First question I have is, would it be more desirable to have a button's name as a global variable? As I tried to add on more dynamic UI, I find myself having tons of global variables at the top referencing each specific UI.

I am guessing I shld either 
1. Shld continue doing this and eventually shift all of them into a globalUI variables file
2. As I decompose the game app logic into smaller helper functions, I should create a new variable that reference a particular UI element again ->e.g 'let newDealBtn = document.querySelector('#buttonThatHasBeenCreatedBefore') ... rest of helper function code' 

This seems more tedious in my opinion and I'm guessing #1 is better.

Second question is - how should I more accurately group helper functions to a related UI? I realised one of the reason why my code becomes more messy and naming +conventions become poorer is due to a clustering of many different levels of helper functions (like helper functions for a helper function) and I get confused lol.

Reply:
Thanks for pointing this out as well!

Reply:
Q1: fewer globals is generally better. We want to reduce the number of globals we have to prevent different parts of our code from accidentally affecting other parts. So your soln #2 is preferred, even though it might be more tedious. 

Q2: Maybe you can write/draw out the dependencies on a piece of paper, then group the related ones together, then put the related ones in individual files :)